src/pseudopeople/noise_functions.py

```diff
@@ -149,13 +153,144 @@ def duplicate_with_guardian(
     randomness_stream: RandomnessStream,
 ) -> pd.DataFrame:
     """
-
-    :param dataset_name:
-    :param dataset_data:
-    :param configuration:
-    :param randomness_stream:
+    Function that duplicates rows of a dataset. Rows that are duplicated fall into one of three groups of
+    dependents that are typically duplicated in administrative data where a dependent lives in a different
+    location then their guardian and they show up in the data at both locations. For simplicity, rows will
+    be only duplicated once for a maximum of two rows per dependent. When a row is duplicated, one row will
+    have the depdent's correct address and the other will have the guardian's address.
+    :param dataset_name: Name of the dataset being noised
+    :param dataset_data: pd.DataFrame that will be noised
+    :param configuration: ConfigTree object containing noise level values. Dict with 3 key groups for duplication
+    :param randomness_stream: RandomnessStream instance to make random selection for noise
     :return:
     """
+
+    # dataset_data = dataset_data.copy()
+    # Get dict of group type and formatted dataframe for that group that should be noised
+    formatted_group_data = {}
+    # Get dataframe for each dependent group to merge with guardians
+    in_households_under_18 = dataset_data.loc[
+        (dataset_data["age"] < 18)
+        & (dataset_data["housing_type"] == "Household")
+        & (dataset_data["guardian_1"].notna())
+    ]
+    in_households_18_to_23 = dataset_data.loc[
+        (dataset_data["age"] >= 18)
+        & (dataset_data["age"] < 24)
+        & (dataset_data["housing_type"] == "Household")
+        & (dataset_data["guardian_1"].notna())
+    ]
+    in_group_quarters_under_24 = dataset_data.loc[
+        (dataset_data["age"] < 24)
+        & (dataset_data["housing_type"] != "Household")
+        & (dataset_data["guardian_1"].notna())
+    ]
+
+    # Helper function to format group dataframe and merging with their dependentss
+    def _merge_depdents_and_guardians(
+        dependents_df: pd.DataFrame, full_data: pd.DataFrame
+    ) -> pd.DataFrame:
+        # Merge dependents with their guardians. We have to merge twice to check
+        # if either guardian is living at a separate location from the dependent.
+        guardian_1s = full_data.loc[
```

Comment:
I'm confused - isn't this adding the "guardian_1_" prefix to EVERYU value in the subset of full_data? Seems like you'd need `full_data.loc[full_data["simulant_id"].isin(full_data["guardian_id"]), "simulant_id"].add_prefix("guardian_1_")`

Reply:
Oh nm, looks like you want more than just the id. Seems like you still want to at least subset to the cols you want. Isn't this adding the prefix to things like age, sex, etc?

Reply:
This could have substantial memory implications.

Reply:
I subset the columns along with the rows now.